architecture-approaches/feature-sliced.md

```diff
@@ -0,0 +1,73 @@
+# Feature sliced
+
+- [What is Feature Sliced Design (FSD)?](<#what-isfeature-sliced-design-(FSD)>)
+- [When do we need use?](#when-do-we-need-use)
+- [Get started](#get-started)
+
+## What is Feature Sliced Design (FSD)?
+
+**“Feature Sliced Design”** — architectural design methodology for Front-End applications, a design approach where a product’s features are divided into separate, standalone components, or “slices”. Each slice is then designed and developed independently, allowing for greater flexibility, modularity, and scalability in the design process. This approach is often used in software development and can be applied to the design of web or mobile applications, as well as other types of software products.
+
+![diagram1](diagram1.webp)
+
+We figured out that idea behind FSD is to break down a complex product into smaller, more manageable pieces in each “Layer”. Each “Layer” might contain other manageable pieces — “Slices”. “Slaces” in turn may have “Segments”.
+
+![diagram2](diagram2.webp)
+
+This can lead to faster development times, better team collaboration, and a more flexible and scalable final product.
+
+Thus we have a set of puzzles that still need to be able to “assemble”. More details are below.
+
+#### Assemble correctly from Childhood!
+
+It's common knowledge among children that puzzles can be pieced together in various ways. Even if the picture doesn't align perfectly, the puzzle pieces stay in place. Isn't that amazing? The mother feels delighted, the father is filled with pride, and the little one realizes that everything is just fine.
+
+Many projects face similar issues such as disorganized imports, duplicated code, and the use of folder locations to define project structure. These problems are commonly referred to as "childhood diseases" and can quickly become a burden. Cleaning up the mess can cost hundreds of man-hours and thousands, if not billions, of dollars. Time is a valuable commodity and should be used efficiently.
+
+> FSD strives to be perfect. It was born with Mission — create a Сontrollable Unidirectional Encapsulated Flow (explicit one-way links).
+
+I imagine a stream that turns into an ocean. A small stream knows nothing about the ocean, but the ocean knows about all the streams, rivers, lakes, and seas that feed it.
+
+> Let’s say more — one river does not know about the existence of the other. This is the prerogative of the seas and oceans.
+
+Technically, It means that the Bottom Layer (for example, shared) knows nothing about all the layers above or next to it. Upper layers (ex. entities) can use all the layers below, like a river that flows into the sea, not vice versa.
```

Comment:
Interesting analogies, but I'd suggest summarizing this text a little bit because it looks like an article (not documentation). What do you think?